dotCMS/src/main/java/com/dotcms/variant/business/web/VariantWebAPI.java

```diff
@@ -0,0 +1,49 @@
+package com.dotcms.variant.business.web;
+
+import com.dotmarketing.util.PageMode;
+import com.liferay.portal.model.User;
+
+/**
+ * WebAPI for {@link com.dotcms.variant.model.Variant}
+ */
+public interface VariantWebAPI {
+
+    /**
+     * Return the current variant ID
+     * @return
+     */
+    String currentVariantId();
+
+    /**
+     * Return the {@link com.dotcms.variant.model.Variant} and {@link com.dotmarketing.portlets.languagesmanager.model.Language}
+     * to render a specific {@link com.dotcms.content.model.Contentlet} according to the versions that
+     * this has.
+     *
+     * @param tryingLang Language to try if not exists any version for this lang try with default
+     * @param identifier {@link com.dotcms.content.model.Contentlet}'s identifier
+     * @param pageMode page mode to render
+     * @param user to check {@link com.dotmarketing.beans.Permission}
+     * @return
+     */
+    RenderContext getRenderContext(final long tryingLang, final String identifier,
+            final PageMode pageMode, final User user);
+
+    class RenderContext {
+        private String currentVariantKey;
```

Comment:
set to final

Reply:
done https://github.com/dotCMS/core/pull/23022/commits/4d91d0bf79d06f7b4cca64afb251f042ac7520a1#diff-45dbdd64c70c9bbcc5509ea0609065408ee20034d258b0ffa416e73c6976f0f5R32